server/pkg/integrationapi/asset_test.go

```diff
@@ -46,4 +51,183 @@ func TestToAssetFile(t *testing.T) {
 		ContentType: lo.ToPtr(""),
 	}
 	assert.Equal(t, e, a)
+
+	assert.Nil(t, ToAssetFile(nil, true))
+}
+
+func Test_NewAsset(t *testing.T) {
+	timeNow := time.Now()
+	name := "aaa"
+	path := "a/aaa"
+	uid := user.NewID()
+	pid := project.NewID()
+	a := asset.New().NewID().Project(pid).Size(100).NewUUID().
+		CreatedByUser(uid).Thread(id.NewThreadID()).CreatedAt(timeNow).MustBuild()
+
+	f1 := asset.NewFile().Name(name).Path(path).ContentType("s").Size(10).Build()
+
+	tests := []struct {
+		name string
+		a    *asset.Asset
+		f    *asset.File
+		url  string
+		all  bool
+		want *Asset
+	}{
+		{
+			name: "success",
+			a:    a,
+			f:    f1,
+			url:  "www.",
+			all:  true,
+			want: &Asset{
+				Name:      lo.ToPtr("aaa"),
+				Id:        a.ID(),
+				Url:       "www.",
+				CreatedAt: timeNow,
+				File: &File{
+					Name:        lo.ToPtr("aaa"),
+					Path:        lo.ToPtr("/a/aaa"),
+					ContentType: lo.ToPtr("s"),
+					Size:        lo.ToPtr(float32(10)),
+				},
+				ContentType: lo.ToPtr("s"),
+				TotalSize:   lo.ToPtr(float32(100)),
+				PreviewType: lo.ToPtr(Unknown),
+				ProjectId:   pid,
+			},
+		},
+		{
+			name: "asset and file input is nil",
+			a:    nil,
+			f:    nil,
+			url:  "www.",
+			all:  false,
+			want: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := NewAsset(tt.a, tt.f, tt.url, tt.all)
+
+			if !reflect.DeepEqual(result, tt.want) {
+				t.Errorf("want %+v, got %+v", tt.want, result)
+			}
+		})
+	}
+}
+
+func TestToAssetArchiveExtractionStatus(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    *asset.ArchiveExtractionStatus
+		expected *AssetArchiveExtractionStatus
+	}{
+		{
+			name:     "Nil input",
+			input:    nil,
+			expected: nil,
+		},
+		{
+			name:     "Status done",
+			input:    lo.ToPtr(asset.ArchiveExtractionStatusDone),
+			expected: lo.ToPtr(AssetArchiveExtractionStatus("done")),
+		},
+		{
+			name:     "Status failed",
+			input:    lo.ToPtr(asset.ArchiveExtractionStatusFailed),
+			expected: lo.ToPtr(AssetArchiveExtractionStatus("failed")),
+		},
+		{
+			name:     "Status in progress",
+			input:    lo.ToPtr(asset.ArchiveExtractionStatusInProgress),
+			expected: lo.ToPtr(AssetArchiveExtractionStatus("in_progress")),
+		},
+		{
+			name:     "Status pending",
+			input:    lo.ToPtr(asset.ArchiveExtractionStatusPending),
+			expected: lo.ToPtr(AssetArchiveExtractionStatus("pending")),
+		},
+		{
+			name:     "Unknown status",
+			input:    lo.ToPtr(asset.ArchiveExtractionStatus("unknown")),
+			expected: nil,
+		},
```

Comment:
```suggestion
		{
			name:     "Status done",
			input:    lo.ToPtr(asset.ArchiveExtractionStatusDone),
			expected: lo.ToPtr(integrationapi.Done),
		},
		{
			name:     "Status failed",
			input:    lo.ToPtr(asset.ArchiveExtractionStatusFailed),
			expected: lo.ToPtr(integrationapi.Failed),
		},
		{
			name:     "Status in progress",
			input:    lo.ToPtr(asset.ArchiveExtractionStatusInProgress),
			expected: lo.ToPtr(integrationapi.InProgress),
		},
		{
			name:     "Status pending",
			input:    lo.ToPtr(asset.ArchiveExtractionStatusPending),
			expected: lo.ToPtr(integrationapi.Pending),
		},
		{
			name:     "Unknown status",
			input:    lo.ToPtr(asset.ArchiveExtractionStatus("unknown")),
			expected: nil,
		},
```
use the defined constants

Reply:
will fix